src/components/Modal/Account.vue

```diff
@@ -1,70 +1,85 @@
+import walletconnectIcon from '@/assets/connectors/walletconnect.png';
+import gnosisIcon from '@/assets/connectors/gnosis.png';
+// import starknetIcon from '@/assets/connectors/starknet.png';
 
-const props = defineProps({ open: Boolean });
-const emit = defineEmits(['login', 'close']);
-const win = window;
+const props = defineProps<{
+  open: boolean;
+}>();
 
-const { open } = toRefs(props);
-const { web3, logout } = useWeb3();
-let step = $ref(null);
+const emit = defineEmits<{
+  (e: 'connect', connector: Connector): void;
+  (e: 'close'): void;
+}>();
+
+const { disconnect, web3Account, ensAddress, chain, isConnected } = useWeb3();
+const { connectors } = useConnect();
+
+let step = ref('');
 
 async function handleLogout() {
-  await logout();
+  disconnect();
   emit('close');
 }
 
-watch(open, () => (step = null));
+function getWalletIcons(id: string) {
+  switch (id) {
+    case 'injected':
+      return metamaskIcon;
+    case 'coinbaseWallet':
+      return coinbaseIcon;
+    case 'walletConnect':
+      return walletconnectIcon;
+    case 'safe':
+      return gnosisIcon;
+    default:
+      return metamaskIcon;
+  }
+}
+
+watch(
+  () => props.open,
+  () => (step.value = '')
+);
 </script>
 
 <template>
   <UiModal :open="open" @close="$emit('close')">
     <template #header>
-      <h3 v-if="!web3.account || step === 'connect'" v-text="'Connect wallet'" />
+      <h3 v-if="!web3Account || step === 'connect'" v-text="'Connect wallet'" />
       <h3 v-else v-text="'Account'" />
     </template>
-    <div v-if="!web3.account || step === 'connect'">
+    <div v-if="!web3Account || step === 'connect'">
       <div class="m-4 space-y-2">
-        <a
-          v-for="(connector, id, i) in connectors"
-          :key="i"
-          target="_blank"
-          class="block"
-          @click="$emit('login', connector.id)"
-        >
+        <template v-for="connector in connectors" :key="connector.id">
           <UiButton
-            v-if="(connector.type === 'injected' && win[connector.root]) || !connector.type"
+            v-if="connector.ready"
             class="button-outline w-full flex justify-center items-center"
+            @click="$emit('connect', connector)"
           >
             <img
-              :src="getConnectorIconUrl(connector.icon)"
+              :src="getWalletIcons(connector.id)"
               height="28"
               width="28"
               class="mr-2 -mt-1"
               :alt="connector.name"
             />
             {{ connector.name }}
           </UiButton>
-        </a>
+        </template>
       </div>
     </div>
     <div v-else>
-      <div v-if="$auth.isAuthenticated.value" class="m-4 space-y-2">
-        <a :href="explorerUrl(web3.network.key, web3.account)" target="_blank" class="block">
+      <div v-if="isConnected" class="m-4 space-y-2">
+        <a :href="explorerUrl(chain.id, web3Account)" target="_blank" class="block">
```

Comment:
@Sekhmet how does `explorerUrl` work for `argentx` if it's not in the networks.json?

Reply:
It has special case: https://github.com/snapshot-labs/sx-ui/blob/c271cf63de0b37b5adfac86f799f1ad422a5e777/src/helpers/utils.ts#L97

Reply:
@Sekhmet yes but `networks[network].explorer` will still be undefined